killswitch.py

```diff
@@ -76,17 +286,28 @@ def __repr__(self) -> str:
         return f'KillSwitchSet(kill_switches={self.kill_switches!r})'
 
 
-KILL_SWITCH_JSON = List[Dict[str, Union[str, List[str]]]]
-KILL_SWITCH_JSON_DICT = Dict[
-    str, Union[
-        str,  # Last updated
-        int,  # Version
-        KILL_SWITCH_JSON  # kills
-    ]
-]
+class BaseSingleKillSwitch(TypedDict):  # noqa: D101
+    reason: str
+
+
+class SingleKillSwitchJSON(BaseSingleKillSwitch, total=False):  # noqa: D101
+    redact_fields: list[str]    # set fields to "REDACTED"
+    delete_fields: list[str]    # remove fields entirely
+    set_fields: dict[str, Any]  # set fields to given data
+
+
+class KillSwitchSetJSON(TypedDict):  # noqa: D101
+    version: str
+    kills: Dict[str, SingleKillSwitchJSON]
+
 
+class KillSwitchJSONFile(TypedDict):  # noqa: D101
+    version: int
+    last_updated: str
+    kill_switches: List[KillSwitchSetJSON]
 
-def fetch_kill_switches(target=DEFAULT_KILLSWITCH_URL) -> Optional[KILL_SWITCH_JSON_DICT]:
+
+def fetch_kill_switches(target=DEFAULT_KILLSWITCH_URL) -> Optional[KillSwitchJSONFile]:
```

Comment:
I'm antsy about this remote-data request not being in a thread, but on reflection:

1. If we have core-code killswitches then we **really** *do* want to load them before running any code they might affect.
2. For now I doubt any plugin is using them, and at very worst we'll take a 10 second hit for core code plus each plugin attempting to load killswitches.

Resolving this to ensure core code always runs with it kill switches is mostly moot because what are we going to do if they're not loaded yet?  Hang around for them to be loaded!

For plugins we could document how plugins can load them in a thread (provide a helper?) and also how to ensure they don't process data until the load/check is performed.

Reply:
Added a oneliner helper and doccumented it. As for core code, yeah, I want us to sit and wait, plugins can just defer anything they send until its there (or do it on-thread and block)